internal/characters/arlecchino/arlecchino.go

```diff
@@ -0,0 +1,130 @@
+package arlecchino
+
+import (
+	"fmt"
+
+	tmpl "github.com/genshinsim/gcsim/internal/template/character"
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/core/info"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/model"
+)
+
+func init() {
+	core.RegisterCharFunc(keys.Arlecchino, NewChar)
+}
+
+type char struct {
+	*tmpl.Character
+	skillDebt             float64
+	skillDebtMax          float64
+	initialDirectiveLevel int
+	chargeEarlyCancelled  bool
+}
+
+func NewChar(s *core.Core, w *character.CharWrapper, _ info.CharacterProfile) error {
+	c := char{}
+	c.Character = tmpl.NewWithWrapper(s, w)
+
+	c.EnergyMax = base.SkillDetails.BurstEnergyCost
+	c.NormalHitNum = normalHitNum
+	c.NormalCon = 3
+	c.BurstCon = 5
+
+	w.Character = &c
+
+	c.chargeEarlyCancelled = false
+
+	return nil
+}
+
+func (c *char) Init() error {
+	c.naBuff()
+	c.passive()
+	c.a1OnKill()
+
+	c.a4()
+
+	c.c2()
+	c.c6()
+	return nil
+}
+
+func (c *char) NextQueueItemIsValid(k keys.Char, a action.Action, p map[string]int) error {
+	if c.chargeEarlyCancelled {
+		// can only early cancel charged attack with Dash or Jump
+		switch a {
+		case action.ActionDash, action.ActionJump:
+			c.chargeEarlyCancelled = false
+		default:
+			return fmt.Errorf("%v: Cannot early cancel Charged Attack with %v", c.CharWrapper.Base.Key, a)
+		}
+	}
+
+	// can use charge without attack beforehand unlike most of the other polearm users
+	if a == action.ActionCharge {
+		return nil
+	}
+	return c.Character.NextQueueItemIsValid(k, a, p)
+}
+
+func (c *char) AnimationStartDelay(k model.AnimationDelayKey) int {
+	switch k {
+	case model.AnimationXingqiuN0StartDelay:
+		return 15
+	case model.AnimationYelanN0StartDelay:
+		return 7
+	default:
+		return c.Character.AnimationStartDelay(k)
+	}
+}
+
+func (c *char) getTotalAtk() float64 {
+	stats, _ := c.Stats()
+	return c.Base.Atk*(1+stats[attributes.ATKP]) + stats[attributes.ATK]
+}
+
+func (c *char) Heal(hi *info.HealInfo) (float64, float64) {
```

Comment:
i'm not sure copying the original implementation is a good idea cuz you always have to rewrite it of there are changes in the original one

Reply:
I believe you discussed this with srl on discord. There overriding the `ModifyHPDebtByAmount()` func only doesn't work, because `CharacterWrapper.FuncThatCallsModifyHP()` won't call `Arlecchino.ModifyHPDebtByAmount()`.

Reply:
I'll address this one in a separate PR since it affects Clorinde implementation as well